### PR TITLE
Allow building with sndio on Linux

### PR DESCRIPTION
**Pull request by ccawley2011**
This currently takes priority over the other audio drivers on Linux, however it'll be skipped if `sndiod` isn't active.

### CODE REVIEW

**Review by AliceLR — COMMENTED** [review]
This seems reasonable to me. I'm going to double check that audioio.h change to OpenBSD/FreeBSD really quick before I approve.

**Review by AliceLR — APPROVED** [review]
* FreeBSD does not have audioio.h. Haven't found evidence that it ever has, and SDL refers to this as "NetBSD/OpenBSD" audio. Nothing of value lost here.
* `audio_info_t` et al. are long gone from OpenBSD sys/audioio.h and if OpenBSD support is kept for this driver, there should be an extra check for some of the removed structs or functions to prevent the current build *breakage*. Yeah, this patch technically *fixes* OpenBSD builds right now (edit: no, the `*)` case needs to be dealt with now).
* Said extra check should probably also be added to the other instance that autodetects BSD audio in the `*)` case.
* Still working on getting autoconf in FreeBSD and NetBSD to output a valid `configure`.

tl;dr I don't have a FreeBSD or OpenBSD environment where the sound_bsd driver would even work currently, and NetBSD has its own version of this driver. This looks OK to me.
